testpar/t_2Gio.c

```diff
@@ -1892,324 +1922,96 @@ dataset_readAll(void)
-    hsize_t     max_dims[MAX_RANK] = {H5S_UNLIMITED, H5S_UNLIMITED}; /* dataset maximum dim sizes */
-    DATATYPE   *data_array1        = NULL;                           /* data buffer */
-    hsize_t     chunk_dims[MAX_RANK];                                /* chunk sizes */
-    hid_t       dataset_pl;                                          /* dataset create prop. list */
-
-    hsize_t start[MAX_RANK];  /* for hyperslab setting */
-    hsize_t count[MAX_RANK];  /* for hyperslab setting */
-    hsize_t stride[MAX_RANK]; /* for hyperslab setting */
-    hsize_t block[MAX_RANK];  /* for hyperslab setting */
-
-    herr_t ret; /* Generic return value */
-    int    mpi_size, mpi_rank;
-
-    MPI_Comm comm = test_comm;
-    MPI_Info info = MPI_INFO_NULL;
+    hid_t       fid;             /* HDF5 file ID */
+    hid_t       fapl_id;         /* File access templates */
+    hid_t       fs;              /* File dataspace ID */
+    hid_t       ms;              /* Memory dataspace ID */
+    hid_t       dataset;         /* Dataset ID */
+    hsize_t     orig_size  = 10; /* Original dataset dim size */
+    hsize_t     new_size   = 20; /* Extended dataset dim size */
+    hsize_t     one        = 1;
+    hsize_t     max_size   = H5S_UNLIMITED; /* dataset maximum dim size */
+    hsize_t     chunk_size = 16384;         /* chunk size */
+    hid_t       dcpl;                       /* dataset create prop. list */
+    int         written[10],                /* Data to write */
+        retrieved[10];                      /* Data read in */
+    int    mpi_size, mpi_rank;              /* MPI settings */
+    int    i;                               /* Local index variable */
+    herr_t ret;                             /* Generic return value */
 
-    filename = GetTestParameters();
+    filename = ((const test_params_t *)params)->filename;
     if (VERBOSE_MED)
-        printf("Extend independent write test on file %s\n", filename);
+        printf("Extend independent write test #2 on file %s\n", filename);
 
     /* set up MPI parameters */
     MPI_Comm_size(test_comm, &mpi_size);
     MPI_Comm_rank(test_comm, &mpi_rank);
 
-    /* setup chunk-size. Make sure sizes are > 0 */
-    chunk_dims[0] = (hsize_t)chunkdim0;
-    chunk_dims[1] = (hsize_t)chunkdim1;
-
-    /* allocate memory for data buffer */
-    data_array1 = (DATATYPE *)malloc((size_t)dim0 * (size_t)dim1 * sizeof(DATATYPE));
-    VRFY((data_array1 != NULL), "data_array1 malloc succeeded");
-
     /* -------------------
      * START AN HDF5 FILE
      * -------------------*/
     /* setup file access template */
-    acc_tpl = create_faccess_plist(comm, info, facc_type);
-    VRFY((acc_tpl >= 0), "");
-
-    /* Reduce the number of metadata cache slots, so that there are cache
-     * collisions during the raw data I/O on the chunked dataset.  This stresses
-     * the metadata cache and tests for cache bugs. -QAK
```

Comment:
Why was this removed?

Reply:
My comment above may not have showed up inline, but every single test function that was removed from this file wasn't being tested in the first place. They were exact or near-exact duplicates of the same functions from testphdf5's t_dset.c and they weren't actually being run by the t_2Gio.c testing; they simply existed in this file for no reason. Since t_2Gio.c previously included testphdf5.h, these functions matched the prototypes declared there for t_dset.c's tests and caused confusion for some code editors.